### ISSUE TITLE
CI: Run tests in parallel on multi-gpu runner

### ISSUE THREAD

**Issue by janeyx99** [pytorch/pytorch#68379]
This is a followup task to https://github.com/pytorch/pytorch/issues/66511.

Once tests are running on one GPU, we can parallelize them to improve time to signal

cc @seemethere @malfet @pytorch/pytorch-dev-infra

**Comment by seemethere** [comment]
This is not currently a priority for the dev infra team, removing from the dev infra backlog

**Comment by seemethere** [comment]
Going to close, please re-open if more context is gathered
